cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2.rs

```diff
@@ -0,0 +1,1027 @@
+			true,
+			1000,
+		));
+
+		assert!(<PenpalB as PenpalBPallet>::ForeignAssets::asset_exists(
+			token_location.clone().try_into().unwrap(),
+		));
+
+		// Register eth on Penpal
+		assert_ok!(<PenpalB as PenpalBPallet>::ForeignAssets::force_create(
+			RuntimeOrigin::root(),
+			eth_location().try_into().unwrap(),
+			penpal_sovereign.clone().into(),
+			true,
+			1000,
+		));
+
+		assert!(<PenpalB as PenpalBPallet>::ForeignAssets::asset_exists(
+			eth_location().try_into().unwrap(),
+		));
+
+		assert_ok!(<PenpalB as Chain>::System::set_storage(
+			<PenpalB as Chain>::RuntimeOrigin::root(),
+			vec![(
+				PenpalCustomizableAssetFromSystemAssetHub::key().to_vec(),
+				Location::new(2, [GlobalConsensus(Ethereum { chain_id: CHAIN_ID })]).encode(),
+			)],
+		));
+	});
+
+	set_up_eth_and_dot_pool(eth_location());
+	set_up_eth_and_dot_pool_on_penpal(eth_location());
+
+	let token_transfer_value = 2_000_000_000_000u128;
+
+	let assets = vec![
+		// the token being transferred
+		NativeTokenERC20 { token_id: token.into(), value: token_transfer_value },
+	];
+
+	let token_asset: xcm::prelude::Asset = (token_location.clone(), token_transfer_value).into();
+	BridgeHubWestend::execute_with(|| {
+		type RuntimeEvent = <BridgeHubWestend as Chain>::RuntimeEvent;
+		let instructions = vec![
+			// Send message to Penpal
+			DepositReserveAsset {
+				// Send the token plus some eth for execution fees
+				assets: Definite(vec![eth_fee_penpal.clone(), token_asset].into()),
+				// Penpal
+				dest: Location::new(1, [Parachain(PARA_ID_B)]),
+				xcm: vec![
+					// Pay fees on Penpal.
+					PayFees { asset: eth_fee_penpal },
+					// Deposit assets to beneficiary.
+					DepositAsset {
+						assets: Wild(AllOf {
+							id: AssetId(token_location.clone()),
+							fun: WildFungibility::Fungible,
+						}),
+						beneficiary: beneficiary.clone(),
+					},
+					SetTopic(H256::random().into()),
+				]
+				.into(),
+			},
+		];
+		let xcm: Xcm<()> = instructions.into();
+		let versioned_message_xcm = VersionedXcm::V5(xcm);
+		let origin = EthereumGatewayAddress::get();
+
+		let message = Message {
+			origin,
+			assets,
+			xcm: versioned_message_xcm.encode(),
+			claimer: Some(claimer_bytes),
+			value: 3_500_000_000_000u128,
+			execution_fee: 1_500_000_000_000u128,
+			relayer_fee: 1_500_000_000_000u128,
+		};
+
+		let (xcm, _) = EthereumInboundQueueV2::do_convert(message, relayer_location).unwrap();
+		let _ = EthereumInboundQueueV2::send_xcm(xcm, AssetHubWestend::para_id().into()).unwrap();
+
+		assert_expected_events!(
+			BridgeHubWestend,
+			vec![RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. }) => {},]
+		);
+	});
+
+	AssetHubWestend::execute_with(|| {
+		type RuntimeEvent = <AssetHubWestend as Chain>::RuntimeEvent;
+		// Check that the assets were issued on AssetHub
+		assert_expected_events!(
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { .. }) => {},
```

Comment:
check asset id - also check for all assets

[example here](https://github.com/paritytech/polkadot-sdk/blob/698d9ae5b32785d3a5a55b770e973bbdb59ad271/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/asset_transfers.rs#L164C5-L167C7) how to use this macro to also check inner values

Reply:
Added field checks for all events.